public/components/data_connections/components/datasource.tsx

```diff
@@ -30,21 +31,29 @@ interface DatasourceDetails {
 }
 
 export const DataSource = (props: any) => {
-  const { dataSource, pplService, http } = props;
+  const { dataSource, http } = props;
```

Comment:
Do you want to use`http` from `CoreRefs` that way you never have to pass it from props.

Reply:
Is there anyway to make `CoreRefs` http always exist?

Reply:
`http` and all other `coreRefs` will be null at the beginning, when the plugin is in start mode. It is safe to assume that when the users is on this page coreRefs elements are not null.

Reply:
Addressed here : [011308f](https://github.com/opensearch-project/dashboards-observability/pull/1008/commits/011308fc5bc7ff3e58140e1af218934b9007854b) - will remove from other child components and from Home in follow up PR. Merging this as is.